tests/test_app/test_data_store.py

```diff
@@ -281,6 +279,30 @@ class DirectoryDataStoreTests(TestCase, DataStoreBaseTests):
     ReadClass = ReadOnlyDirectoryDataStore
     WriteClass = WritableDirectoryDataStore
 
+    def test_write_class_source_create_delete(self):
+        # tests the case when the directory has other different suffixes to self.suffix
+        os.mkdir("delme_dir")
```

Comment:
This approach to creating a working directory is fragile, for instance if it was not cleaned up by a previous test. Use the
`from tempfile import TemporaryDirectory` approach (e.g. see `test_app.test_io.test_findall_zip`). You can then create the `delme_dir` inside the temporary directory and the `tempfile` module will deal with cleanup.